src/gui/tray/Window.qml

```diff
@@ -624,56 +618,184 @@ Window {
                         }
                     }
                 }
-                Button {
-                    id: shareButton
+                RowLayout {
+                    id: activityActionsLayout
                     anchors.right: activityItem.right
-
-                    Layout.preferredWidth: (path === "") ? 0 : parent.height
-                    Layout.preferredHeight: parent.height
+                    spacing: 0
                     Layout.alignment: Qt.AlignRight
-                    flat: true
-                    hoverEnabled: true
-                    visible: (path === "") ? false : true
-                    display: AbstractButton.IconOnly
-                    icon.source: "qrc:///client/theme/share.svg"
-                    icon.color: "transparent"
-                    background: Rectangle {
-                        color: parent.hovered ? Style.lightHover : "transparent"
+
+                    function actionButtonIcon(actionIndex) {
+                        if (links[actionIndex]._verb === "WEB" && (objectType === "chat" || objectType === "call")) {

```

Comment:
I'd store the verb in an intermediate variably in the beginning.

Shouldn't that be model.links and model.objectType by the way? Unsure where those are coming from right now.